src/db/sysdb_views.c

```diff
@@ -845,17 +1039,37 @@ errno_t sysdb_apply_default_override(struct sss_domain_info *domain,
     struct ldb_message_element el_del = { 0, SYSDB_SSH_PUBKEY, 0, NULL };
     struct sysdb_attrs del_attrs = { 1, &el_del };
 
-    if (override_attrs == NULL) {
-        /* nothing to do */
-        return EOK;
-    }
-
     tmp_ctx = talloc_new(NULL);
     if (tmp_ctx == NULL) {
         DEBUG(SSSDBG_OP_FAILURE, "talloc_new failed.\n");
         return ENOMEM;
     }
 
+    if (override_attrs == NULL) {
+        /* No overrides exist for the user, check templates */
+        if (domain->template_homedir != NULL || domain->template_shell != NULL
+            || global_template_homedir != NULL || global_template_shell != NULL) {
+
+            override_attrs = sysdb_new_attrs(tmp_ctx);
+            if (override_attrs == NULL) {
+                DEBUG(SSSDBG_OP_FAILURE, "sysdb_new_attrs failed.\n");
+                return ENOMEM;
+                goto done;
+            }
+
+            ret = sysdb_add_template_values(override_attrs, domain, global_template_homedir,
+                                            global_template_shell, ldb_dn_get_linearized(obj_dn));
+            if (ret != EOK) {
+                DEBUG(SSSDBG_MINOR_FAILURE, "Error adding template values: [%s]\n",
+                                            strerror(ret));
+                goto done;
+            }
+        /* No templates, nothing to do */
+        } else {
+            return EOK;
```

Comment:
`tmp_ctx` is not being released.

Should be
```
ret = EOK;
goto done;
```

Reply:
Fixed.